git_wrapper/repo.py

```diff
@@ -167,3 +169,18 @@ def remote(self, new_remote):
             :param git_wrapper.remote.GitRemote new_remote: An already constructed GitRemote object to use
         """
         self._remote = new_remote
+
+    @property
+    def branch(self):
+        """Return object to act on the repo's branches"""
+        if not self._branch:
+            self._branch = GitBranch(git_repo=self, logger=self.logger)
```

Comment:
Minor thought, and I could be wrong here - if we are calling self._branch, is the setter ever actually called?

Reply:
I guess the setter is relevant for external callers. The setter used to be called here in a previous patchset, I don't really mind either way.

Reply:
k,it can be addressed later, if needed